lib/rules/require-matching-function-name.js

```diff
@@ -96,11 +96,13 @@ module.exports.prototype = {
 };
 
 function checkForMember(assignment, skip, errors) {
-    if (skip(assignment.left.property.name, assignment.right)) {
+    var _left = assignment.left;
+    var _name = _left.property ? (_left.property.name || _left.property.value) : _left.name;
```

Comment:
There are three options of assignments:
```
var simpleVar = function () {};
obj.propertyName = function () {};
obj['propertyValue'] = function () {};
```
And _left property only exists for the second and the third line.

I'll add here comments.